app/src/main/java/org/oppia/android/app/customview/ContinueButtonView.kt

```diff
@@ -0,0 +1,54 @@
+package org.oppia.android.app.customview
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.animation.AccelerateInterpolator
+import android.view.animation.AnimationUtils
+import androidx.lifecycle.Observer
+import org.oppia.android.R
+import org.oppia.android.app.player.state.itemviewmodel.ContinueInteractionViewModel
+
+class ContinueButtonView @JvmOverloads constructor(
+  context: Context,
+  attrs: AttributeSet? = null,
+  defStyleAttr: Int = R.style.StateButtonActive
+) : androidx.appcompat.widget.AppCompatButton(context, attrs, defStyleAttr) {
+
+  private lateinit var viewModel: ContinueInteractionViewModel
+  private var isAnimationTimerFinished = false
+
+  fun setViewModel(viewModel: ContinueInteractionViewModel) {
+    this.viewModel = viewModel
+    viewModel.subscribeToCurrentState()
+    viewModel.animateContinueButton.observeForever(continueButtonAnimationObserver)
+  }
+
+  private val continueButtonAnimationObserver = Observer<Boolean> {
+    if (it) {
+      startAnimating()
+    } else {
+      this.clearAnimation()
+      isAnimationTimerFinished = false
+    }
+  }
+
+  private fun startAnimating() {
+    val animation = AnimationUtils.loadAnimation(context, R.anim.expand)
```

Comment:
We should name the animation something more contextually specific to what it's being used for (e.g. maybe "pulse_button_size" or something similar?).

Reply:
done